crates/polars-core/src/series/mod.rs

```diff
@@ -157,17 +157,17 @@ impl Series {
     }
 
     pub fn clear(&self) -> Series {
-        // Only the inner of objects know their type, so use this hack.
-        #[cfg(feature = "object")]
-        if matches!(self.dtype(), DataType::Object(_, _)) {
-            return if self.is_empty() {
-                self.clone()
-            } else {
-                let av = self.get(0).unwrap();
-                Series::new(self.name(), [av]).slice(0, 0)
-            };
+        match self.dtype() {
+            #[cfg(feature = "object")]
+            DataType::Object(_, _) => {
+                if self.is_empty() {
+                    self.clone()
+                } else {
+                    self.slice(0, 0)
```

Comment:
We should use `take` because slice doesn't clear the memory underneath.

Reply:
@ritchie46 what's the best syntax for using `take` with empty input? Looks like I need to construct an empty `ChunkedArray`, is there a cleaner way than doing:

```rust
self.take(&UInt32Chunked::new_vec("", vec![])).unwrap()
```

? I could use `unsafe { self.take_unchecked(&[]) }` but that requires unsafe.

Reply:
`self.take(&UInt32Chunked::new_vec("", vec![])).unwrap()`

Is much cleaner than `unsafe` ;). Maybe something with `new_empty`...